### DISCUSSION TITLE
Constraint issues (node_state) used in Spine

### DISCUSSION THREAD

**Discussion by shuo1990** [spine-tools/SpineOpt.jl#961] [Support]
Hi,

I have a model, as seen in the Figure.

There are two units (V2H charger and EV availability) and three nodes (Household, EV, EV usage).

I have already set node_state_min (10% SOC) in the node named 'EV'. It is adopted for the route of EV- EV availability - EV usage.

How can I separately set another minimum constraint (30% SOC) for the other route EV - V2H charger - Household?

Kind regards
Shuo
![Issue_constraint_Shuo](https://github.com/spine-tools/Spine-Toolbox/assets/78427232/212dd464-05fd-4158-8d70-7acb5931f5ca)

**Comment by soininen** [comment]
This looked more like SpineOpt issue than Spine Toolbox so I moved it here.

**Comment by DillonJ** [comment]
Hi @shuo1990. It's not clear to me what you are trying to do. The state of charge (or `node_state` in SpineOpt terms) is a variable associated with the node. Any flows into that node can increase the `node_state` and flows out of the node can reduce the `node_state`. The `node_state` variable represents the energy stored in the node at a point in time and specifying a `node_state_min` will create a lower bound constraint on that variable. The energy stored in the node (`node_state`) is specific only to the node and is not specific to the energy flows to any units connected to that node, however any flows between the node and units will impact that `node_state`. Perhaps you want to set a minimum flow to the EV availability and V2H Charger units using the `min_unit_flow` parameter on the `unit__to_node` / `unit__from_node` relationships?

**Comment by DillonJ** [comment]
@shuo1990 inferring from the duplicate issue #954, you want to prevent discharge from the EV node. Then, you want to constrain the flows from the node rather than the state of charge. I would use either `min_unit_flow` or `unit_capacity` to set lower/upper bounds on the discharge flow as appropriate - perhaps using time patterns or time series.

**Comment by clizbe** [comment]
@DillonJ Thanks for the quick response. I'm converting this to a Support Discussion, so please continue the convo there.
